### ISSUE TITLE
Ska vi inkludera kommentarer i den gemensamma makefilen för electrotest?

### ISSUE THREAD

**Issue by tojo0084** [tojo0084/Lab6#15]
Angående redovisningen kan man läsa bl.a. följande:
"
Redovisa en beskrivning av hur du länkat in alla 3 bibliotek och använt det i ett huvudprogram enligt ovan. Beskriv också vilka du jobbat med.
Beskriv vilka kommandon och växlar du använder för att kompilera biblioteken och hur du sedan använder biblioteken i ditt huvudprogram...
Del 2 i rapporten får gärna vara identiska.
"
Hur tänker ni er att redovisa del 2 som "får gärna vara identiska"?
Beskriva växlarna i ett eget word-dokument eller gemensamma kommentarer i electrotest makefilen?

**Comment by agnon** [comment]
Kan vi inte använda README:n för att redovisa detta? Då kan vi laborera
fram den tesammans :)

2017-05-10 23:29 GMT+02:00 tojo0084 <notifications@github.com>:

**Comment by tojo0084** [comment]
Ja, readme-filen kan nog vara lämplig att använda för en övergripande beskrivning av programmet men detaljerna om "växlar du använder för att kompilera biblioteken" passar väl bättre som kommentarer i makefilen där de används ?
Eller ?
Typ så här alltså i makefilen:
 - - - - - - - - - - - - - - 
CC=gcc
CFLAGS=-std=c11 -Wall -Wextra -Werror -pedantic -g
`# -std=c11 används för att ...`
`# -Wall används för att ...`
`# -Wextra används för att ...`
`# -pedantic -g används för att ...`
`# -g används för att ...`

...

install:
	install -m 755 $(LIB_DIR)/*.so $(DEST_LIB_DIR)
	install -m 755 $(BIN) $(DEST_BIN_DIR)
	@ldconfig
`# @ldconfig anropas för att ...`

...

$(BIN): $(OBJ_DIR)/electrotest.o
	$(CC) $(CFLAGS) -o $@ $^ -L$(LIB_DIR) $(LDFLAGS) -Wl,-rpath,'$$ORIGIN/$(LIB_DIR)'
`# [förklaring av $@]`
`# [förklaring av $^]`
`# [förklaring av -L$]`
`# [förklaring av -Wl]`
`# [förklaring av -rpath]`
`# [förklaring av $$ORIGIN]`

...
- - - - - - - - - - - - - -

**Comment by agnon** [comment]
Jag kan ta på mig att fixa en förklaring av makefile (electrotest då, som är gemensam) i readme:n ikväll när barnen har somnat :)

**Comment by tojo0084** [comment]
Okej det låter bra :-)
